reconcile/test/test_ocm_clusters.py

```diff
@@ -1,305 +1,464 @@
+    assert n.console_url == osd_cluster_fxt[SPEC_ATTR_CONSOLE_URL]
+
+
+def test_ocm_spec_population_osd_with_extra(osd_cluster_fxt):
+    osd_cluster_fxt["spec"]["extra_attribute"] = True
+    with pytest.raises(ValidationError):
+        OCMSpec(**osd_cluster_fxt)
+
+
+def test_get_ocm_cluster_update_spec_no_changes(
+    ocm_mock, ocm_osd_cluster_spec: OCMSpec
+):
+    current_spec = ocm_osd_cluster_spec
+    desired_spec = ocm_osd_cluster_spec
+    upd, err = occ.get_cluster_ocm_update_spec(
+        ocm_mock, "cluster1", current_spec, desired_spec
+    )
+    assert (upd, err) == ({}, False)
+
+
+def test_get_ocm_cluster_update_spec_network_banned(
+    ocm_mock, ocm_osd_cluster_spec: OCMSpec
+):
+    current_spec = ocm_osd_cluster_spec
+    desired_spec = current_spec.copy(deep=True)
+    desired_spec.network.vpc = "0.0.0.0/0"
+    _, err = occ.get_cluster_ocm_update_spec(
+        ocm_mock, "cluster1", current_spec, desired_spec
+    )
+    assert err is True
+
+
+@typing.no_type_check
+def test_get_ocm_cluster_update_spec_allowed_change(
+    ocm_mock, ocm_osd_cluster_spec: OCMSpec
+):
+    current_spec = ocm_osd_cluster_spec
+    desired_spec = current_spec.copy(deep=True)
+    desired_spec.spec.storage = 2000
+    upd, err = occ.get_cluster_ocm_update_spec(
+        ocm_mock, "cluster1", current_spec, desired_spec
+    )
+    assert (upd, err) == ({ocmmod.SPEC_ATTR_STORAGE: 2000}, False)
+
+
+def test_get_ocm_cluster_update_spec_not_allowed_change(
+    ocm_mock, ocm_osd_cluster_spec: OCMSpec
+):
+    current_spec = ocm_osd_cluster_spec
+    desired_spec = current_spec.copy(deep=True)
+    desired_spec.spec.multi_az = not desired_spec.spec.multi_az
+    upd, err = occ.get_cluster_ocm_update_spec(
+        ocm_mock, "cluster1", current_spec, desired_spec
+    )
+    assert (upd, err) == (
+        {ocmmod.SPEC_ATTR_MULTI_AZ: desired_spec.spec.multi_az},
+        True,
+    )
+
+
+def test_get_ocm_cluster_update_spec_disable_uwm(
+    ocm_mock, ocm_osd_cluster_spec: OCMSpec
+):
+    current_spec = ocm_osd_cluster_spec
+    desired_spec = current_spec.copy(deep=True)
+    desired_spec.spec.disable_user_workload_monitoring = (
+        not desired_spec.spec.disable_user_workload_monitoring
+    )
+    upd, err = occ.get_cluster_ocm_update_spec(
+        ocm_mock, "cluster1", current_spec, desired_spec
+    )
+    assert (upd, err) == (
+        {
+            ocmmod.SPEC_ATTR_DISABLE_UWM: desired_spec.spec.disable_user_workload_monitoring
+        },
+        False,
+    )
+
+
+def test_noop_dry_run(queries_mock, ocmmap_mock, ocm_mock, cluster_updates_mr_mock):
+    with pytest.raises(SystemExit):
+        occ.run(False)
+    # If get has not been called means no action has been performed
+    _post, _patch = ocm_mock
+    assert _post.call_count == 0
+    assert _patch.call_count == 0
+    assert cluster_updates_mr_mock.call_count == 0
+
+
+@patch.object(OCM, "_get_json", autospec=True)
```

Comment:
The standard way to patch is to use `mocker` as per [this doc](https://service.pages.redhat.com/dev-guidelines/docs/appsre/python-testing-guide/#pytest-best-practices).

The API of the patched object is the same, so the only thing that should change is adding `mocker` to the arguments for the test function, and then adding a line for: `json = mocker.patch.object(OCM, "_get_json", autospec=True)`

I'll not repeat this comment on the rest of the cases below.

Reply:
I've replaced all the `@patch.object` annotations with a fixture.